sql/data_sql_warehouse.go

```diff
@@ -31,15 +32,37 @@ func DataSourceWarehouse() *schema.Resource {
 
 	return common.DataResource(SQLWarehouseInfo{}, func(ctx context.Context, e interface{}, c *common.DatabricksClient) error {
 		data := e.(*SQLWarehouseInfo)
-		err := c.Get(ctx, fmt.Sprintf("/sql/warehouses/%s", data.ID), nil, data)
+		var id string
+		if data.ID == "" && data.Name == "" {
+			return fmt.Errorf("either 'id' or 'name' should be provided")
+		}
+		endpointsAPI := NewSQLEndpointsAPI(ctx, c)
+		selected := []DataSource{}
+		dataSources, err := endpointsAPI.listDataSources()
 		if err != nil {
 			return err
 		}
-		endpointsAPI := NewSQLEndpointsAPI(ctx, c)
-		data.DataSourceID, err = endpointsAPI.ResolveDataSourceID(data.ID)
+		for _, source := range dataSources {
+			log.Printf("[DEBUG] source=%v", source)
+			if data.Name != "" && source.Name == data.Name {
+				selected = append(selected, source)
+			} else if data.ID != "" && source.EndpointID == data.ID {
+				selected = append(selected, source)
+				break
+			}
+		}
+		if len(selected) == 0 {
+			return fmt.Errorf("can't find SQL warehouse with the name '%s'", data.Name)
```

Comment:
suggestion: let's tune the error message based on whether ID or Name was specified.